files/image_config/pcie-check/pcie-check.sh

```diff
@@ -0,0 +1,58 @@
+#! /bin/bash
+## Check the platform PCIe device presence and status
+
+VERBOSE="no"
+RESULTS="PCIe Device Checking All Test"
+PCIE_CHK_CMD=`sudo pcieutil pcie-check |grep "$RESULTS"`
+EXPECTED="PCIe Device Checking All Test ----------->>> PASSED"
+MAX_WAIT_TIMEOUT=15
+
+function debug()
+{
+    /usr/bin/logger "$0 : $1"
+    if [[ x"${VERBOSE}" == x"yes" ]]; then
+        echo "$(date) $0: $1"
+    fi
+}
+
+function check_and_rescan_pcie_devices()
+{
+    PLATFORM=`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`
+    if [ ! -f /usr/share/sonic/device/$PLATFORM/plugins/pcie.yaml ]; then
+        debug "pcie.yaml does not exist! can't check pcie status!"
+        exit
+    fi
+
+    begin=$SECONDS
+    end=`expr $begin + $MAX_WAIT_TIMEOUT`
+    rescan_time=$((MAX_WAIT_TIMEOUT/2))
```

Comment:
So now we only rescan once at most (after waiting 7.5 seconds) instead of 15 times? Is this correct?

Reply:
@jleveque Yes, correct.